src/Databases/DatabaseFilesystem.cpp

```diff
@@ -0,0 +1,189 @@
+#include <Databases/DatabaseFilesystem.h>
+
+#include <IO/Operators.h>
+#include <IO/WriteBufferFromString.h>
+#include <Interpreters/Context.h>
+#include <Parsers/ASTCreateQuery.h>
+#include <Parsers/ASTFunction.h>
+#include <Parsers/ASTLiteral.h>
+#include <Parsers/ParserCreateQuery.h>
+#include <Parsers/parseQuery.h>
+#include <Storages/IStorage.h>
+#include <TableFunctions/TableFunctionFactory.h>
+
+#include <filesystem>
+
+namespace fs = std::filesystem;
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int LOGICAL_ERROR;
+    extern const int UNKNOWN_TABLE;
+    extern const int DATABASE_ACCESS_DENIED;
+    extern const int BAD_ARGUMENTS;
+}
+
+DatabaseFilesystem::DatabaseFilesystem(const String & name_, const String & path_, ContextPtr context_)
+    : IDatabase(name_), WithContext(context_->getGlobalContext()), path(path_), log(&Poco::Logger::get("DatabaseFileSystem(" + name_ + ")"))
+{
+    fs::path user_files_path;
+    if (context_->getApplicationType() != Context::ApplicationType::LOCAL)
+        user_files_path = fs::canonical(fs::path(getContext()->getUserFilesPath()));
+
+    if (fs::path(path).is_relative())
+        path = user_files_path / path;
```

Comment:
need to add 
```
if (fs::path(path).is_relative())
         path = user_files_path / path;
else if (!pathStartsWith(path, user_files_path))
         throw Exception(ErrorCodes::BAD_ARGUMENTS, "Path must be inside user-files path ({})", user_files_path);
```
(function `pathStartsWith` is from `filesystemHelpers.h`)

Reply:
fixed